libcontainer/utils/utils_unix.go

```diff
@@ -364,3 +365,27 @@ func Openat(dir *os.File, path string, flags int, mode uint32) (*os.File, error)
 	}
 	return os.NewFile(uintptr(fd), dir.Name()+"/"+path), nil
 }
+
+// RetryOnEINTR takes a function that returns an error and calls it until it the error returned is
+// not EINTR.
+func RetryOnEINTR(fn func() error) error {
```

Comment:
I'm very much against adding new public methods to runc packages.

Guess we need to create a package in `libcontainer/internal`. Maybe call it `linux` or `unix`?

Reply:
Yeah, any new helpers should be moved to internal. Maybe we should just try to move all of utils in one go as well? All the stuff in there is really runc-specific and I really hope nobody is using it because I have historically had less than zero cares about breaking those APIs.

Reply:
Oh, great point! I'll create this in internal, in case someone has an issue moving all, and we can move the rest (and discuss that) in another PR

Reply:
Moved to libcontainer/internal/unix-utils. Thanks!